diff --git a/android/CMakeLists.txt b/android/CMakeLists.txt
index cc4cbfea0fa561bab2d28c82f090825b3e5655bf..7b18e1ceb1c6bd1263a0942a8d372a110ba5efc9 100644
--- a/android/CMakeLists.txt
+++ b/android/CMakeLists.txt
@@ -79,10 +79,17 @@ if(${JS_RUNTIME} STREQUAL "hermes")
 
   string(APPEND CMAKE_CXX_FLAGS " -DJS_RUNTIME_HERMES=1")
 
-  target_link_libraries(
-    ${PACKAGE_NAME}
-    hermes-engine::libhermes
-  )
+  if(ReactAndroid_VERSION_MINOR GREATER_EQUAL 82)
+    target_link_libraries(
+      ${PACKAGE_NAME}
+      hermes-engine::hermesvm
+    )
+  else()
+    target_link_libraries(
+      ${PACKAGE_NAME}
+      hermes-engine::libhermes
+    )
+  endif()
 
   if(${HERMES_ENABLE_DEBUGGER})
     if(ReactAndroid_VERSION_MINOR GREATER_EQUAL 76)
